public/app/features/users/SignupInvited.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { FC, useState } from 'react';
+import { hot } from 'react-hot-loader';
+import { connect, MapStateToProps, MapDispatchToProps } from 'react-redux';
+import { StoreState } from 'app/types';
+import { updateLocation } from 'app/core/actions';
+import { UrlQueryValue, getBackendSrv } from '@grafana/runtime';
+import { Forms } from '@grafana/ui';
+import { useAsync } from 'react-use';
+import Page from 'app/core/components/Page/Page';
+import { contextSrv } from 'app/core/core';
+import { getConfig } from 'app/core/config';
+
+interface ConnectedProps {
+  code?: UrlQueryValue;
+}
+
+interface DispatchProps {
+  updateLocation: typeof updateLocation;
+}
+
+interface FormModel {
+  email: string;
+  name?: string;
+  username: string;
+  password?: string;
+}
+
+const navModel = {
+  main: {
+    icon: 'gicon gicon-branding',
+    text: 'Invite',
+    subTitle: 'Register your Grafana account',
+    breadcrumbs: [{ title: 'Login', url: 'login' }],
+  },
+  node: {
+    text: '',
+  },
+};
+
+const SingupInvitedPageUnconnected: FC<DispatchProps & ConnectedProps> = ({ code }) => {
+  const [initFormModel, setInitFormModel] = useState<FormModel>();
+  const [greeting, setGreeting] = useState<string>();
+  const [invitedBy, setInvitedBy] = useState<string>();
+  useAsync(async () => {
```

Comment:
useAsync returns state. I think you can wait with rendering the form until this has initialised